docs: refresh testing_integration.rst (#53338)
ansible/ansible: docs/docsite/rst/dev_guide/testing_integration.rst

Configuration
=============

ansible-test command
--------------------

> The example below assumes ``test/runner`` is in your ``$PATH``. An easy way to achieve that

Reviewer: ```suggestion
The example below assumes ``bin`` is in your ``$PATH``. An easy way to achieve that
```

This was written before `ansible-test` was added to `bin`. At the time it was only available as `test/runner/ansible-test`, but is now also available from the preferred location of `bin/ansible-test`. Sourcing `hacking/env-setup` makes the one under `bin` available.

is to initialize your environment with the ``env-setup`` command::

    source hacking/env-setup
    ansible-test --help

You can also call ``ansible-test`` with the full path::

>     test/runner/ansible-test --help

Reviewer: ```suggestion
    bin/ansible-test --help
```

integration_config.yml
----------------------

> 

Reviewer: The section below that references `credentials.template` is out-of-date as well. I'd probably replace the last sentence:

> For more information about supported credentials, refer to ``credentials.template``.

with:

> For more information about supported credentials, refer to the various ``cloud-config-*.template`` files in the ``test/integration/`` directory.

Prerequisites
=============

> The tests will assume things like hg, svn, and git are installed, and in path.  Some tests

Reviewer: ```suggestion
Some tests assume things like hg, svn, and git are installed, and in path.  Some tests
```

Run as follows for all POSIX platform tests executed by our CI system::

>     ansible-test integration --docker fedora29 -v shippable/

Reviewer: Sorry, I should have been more clear on my previous feedback. You'll probably want to add a brief section mentioning use of `source hacking/env-setup` (as is found in several other places in our docs, although not currently in this file), so that `ansible-test` is in the path. Then update this example, as you've done, along with all the others in this file to use `ansible-test` instead of `test/runner/ansible-test`.

Author: Is it better this way: https://github.com/ansible/ansible/pull/53338/commits/205b960a662e6fccb9747605cfdf3a3d483c97b3

As way tempted to just add `source hacking/env-setup` before every ansible-test example. This may be enough.

>     test/runner/ansible-test integration --docker fedora29 -v shippable/posix/

Reviewer: The `ansible-test` invocations could be simplified if the docs mentioned sourcing `hacking/env-setup`, since that puts `ansible-test` in `PATH`.

```suggestion
    ansible-test integration --docker fedora29 -v shippable/
```

Run as follows for all POSIX platform tests executed by our CI system::

    ansible-test integration --docker fedora29 -v shippable/

You can target a specific tests as well, such as for individual modules::

    ansible-test integration -v ping

Use the following command to list all the available targets::

    ansible-test integration --list-targets

.. note:: Bash users

>    If you use ``Bash`` with ``argcomplete``, obtain a full list by doing: ``test/runner/ansible-test integration <tab><tab>``

Reviewer: ```suggestion
   If you use ``bash`` with ``argcomplete``, obtain a full list by doing: ``ansible-test integration <tab><tab>``
```

Run as follows for all POSIX platform tests executed by our CI system::

    ansible-test integration --docker fedora29 -v shippable/

You can target a specific tests as well, such as for individual modules::

    ansible-test integration -v ping

Use the following command to list all the available targets::

    ansible-test integration --list-targets

.. note:: Bash users

>    If you use ``bash`` with ``argcomplete``, obtain a full list by doing: ``test/runner/ansible-test integration <tab><tab>``

Reviewer: ```suggestion
   If you use ``bash`` with ``argcomplete``, obtain a full list by doing: ``ansible-test integration <tab><tab>``
```

Run the Windows tests executed by our CI system::

>     ansible-test windows-integration -v windows/ci/

Reviewer: ```suggestion
    ansible-test windows-integration -v shippable/
```

To run all CI integration test targets for POSIX platforms in a Ubuntu 16.04 container::

>     test/runner/ansible-test integration --docker -v shippable/

Reviewer: ```suggestion
    test/runner/ansible-test integration --docker ubuntu1604 -v shippable/
```
